File: backend/app/schemas/subscription.py
"""Pydantic models for subscription and usage summary endpoints."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PlanSummaryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    code: str
    name: str
    monthly_story_limit: int | None
    max_pages_per_story: int | None
    image_quality_mode: str | None
    voice_clone_limit: int | None
    monthly_audio_chars_limit: int | None
    price_cents: int


class UsageMetricResponse(BaseModel):
    used: int
    limit: int | None
    remaining: int | None
    unit: str


class SubscriptionSummaryResponse(BaseModel):
    id: uuid.UUID
    status: str
    current_period_start: datetime | None
    current_period_end: datetime | None
    plan: PlanSummaryResponse
    usage: dict[str, UsageMetricResponse]

File: frontend/src/services/subscription.ts
import { api } from './api'

export interface UsageMetric {
  used: number
  limit: number | null
  remaining: number | null
  unit: string
}

export interface PlanSummary {
  id: string
  code: string
  name: string
  monthly_story_limit: number | null
  max_pages_per_story: number | null
  image_quality_mode: string | null
  voice_clone_limit: number | null
  monthly_audio_chars_limit: number | null
  price_cents: number
}

export interface SubscriptionSummary {
  id: string
  status: string
  current_period_start: string | null
  current_period_end: string | null
  plan: PlanSummary
  usage: Record<string, UsageMetric>
}

export async function getSubscriptionSummary(): Promise<SubscriptionSummary> {
  const { data } = await api.get<SubscriptionSummary>('/subscription')
  return data
}
